diff --git a/erts/emulator/beam/global.h b/erts/emulator/beam/global.h
index f918cbfe6b0613667fb7a1519962323ec0a275fe..bc72a4a68c020d66d1e5cf7f453842a26d3b0daa 100644
--- a/erts/emulator/beam/global.h
+++ b/erts/emulator/beam/global.h
@@ -225,7 +225,9 @@ struct erts_driver_t_ {
     DE_Handle *handle;
     erts_mtx_t *lock;
     ErlDrvEntry *entry;
-    ErlDrvData (*start)(ErlDrvPort port, char *command, SysDriverOpts* opts);
+    ErlDrvData (*start)(ErlDrvPort port, char *command);
+    ErlDrvData (*system_start)(ErlDrvPort port, char *command,
+                               SysDriverOpts* opts);
     void (*stop)(ErlDrvData drv_data);
     void (*finish)(void);
     void (*flush)(ErlDrvData drv_data);
diff --git a/erts/emulator/beam/io.c b/erts/emulator/beam/io.c
index 24468adbe35712bdb751349b0484df3c3ead76c1..fdf409aad2cb40415bc70ccc0459eeeea238b61d 100644
--- a/erts/emulator/beam/io.c
+++ b/erts/emulator/beam/io.c
@@ -93,7 +93,7 @@ static erts_atomic64_t bytes_in;
 static erts_atomic64_t bytes_out;
 
 static void deliver_result(Port *p, Eterm sender, Eterm pid, Eterm res);
-static int init_driver(erts_driver_t *, ErlDrvEntry *, DE_Handle *);
+static int init_driver(erts_driver_t *, ErlDrvEntry *, DE_Handle *, int);
 static void terminate_port(Port *p);
 static void pdl_init(void);
 static int driver_failure_term(ErlDrvPort ix, Eterm term, int eof);
@@ -685,7 +685,7 @@ erts_open_driver(erts_driver_t* driver,	/* Pointer to driver. */
                                  &opts->high_msgq_watermark);
 
     error_number = error_type = 0;
-    if (driver->start) {
+    if (driver->start || driver->system_start) {
         ERTS_MSACC_PUSH_STATE_M();
 	if (ERTS_IS_P_TRACED_FL(port, F_TRACE_SCHED_PORTS)) {
 	    trace_sched_ports_where(port, am_in, am_open);
@@ -710,7 +710,13 @@ erts_open_driver(erts_driver_t* driver,	/* Pointer to driver. */
         }
 #endif
 
-	drv_data = (*driver->start)(ERTS_Port2ErlDrvPort(port), name, opts);
+	if (driver->system_start) {
+	    drv_data = (*driver->system_start)(ERTS_Port2ErlDrvPort(port),
+					     name, opts);
+	}
+	else {
+	    drv_data = (*driver->start)(ERTS_Port2ErlDrvPort(port), name);
+	}
 	if (((SWord) drv_data) == -1)
 	    error_type = -1;
 	else if (((SWord) drv_data) == -2) {
@@ -3028,10 +3034,10 @@ void erts_init_io(int port_tab_size,
     erts_tsd_set(driver_list_lock_status_key, (void *) 1);
     erts_rwmtx_rwlock(&erts_driver_list_lock);
 
-    init_driver(&fd_driver, &fd_driver_entry, NULL);
-    init_driver(&spawn_driver, &spawn_driver_entry, NULL);
+    init_driver(&fd_driver, &fd_driver_entry, NULL, 1);
+    init_driver(&spawn_driver, &spawn_driver_entry, NULL, 1);
 #ifndef __WIN32__
-    init_driver(&forker_driver, &forker_driver_entry, NULL);
+    init_driver(&forker_driver, &forker_driver_entry, NULL, 1);
 #endif
     erts_init_static_drivers();
     for (dp = driver_tab; dp->de != NULL; dp++)
@@ -7551,7 +7557,8 @@ no_stop_select_callback(ErlDrvEvent event, void* private)
      ((DE)->major_version == (MAJOR) && (DE)->minor_version >= (MINOR)))
 
 static int
-init_driver(erts_driver_t *drv, ErlDrvEntry *de, DE_Handle *handle)
+init_driver(erts_driver_t *drv, ErlDrvEntry *de, DE_Handle *handle,
+	    int is_system_driver)
 {
     drv->name_atom = erts_atom_put((byte*)de->driver_name,
                                    sys_strlen(de->driver_name),
@@ -7574,7 +7581,14 @@ init_driver(erts_driver_t *drv, ErlDrvEntry *de, DE_Handle *handle)
     }
     drv->entry = de;
 
-    drv->start = de->start;
+    if (is_system_driver) {
+	drv->start = NULL;
+	drv->system_start = de->start;
+    }
+    else {
+	drv->start = (ErlDrvData (*)(ErlDrvPort, char *)) de->start;
+	drv->system_start = NULL;
+    }
     drv->stop = de->stop;
     drv->finish = de->finish;
     drv->flush = de->flush;
@@ -7655,7 +7669,7 @@ int erts_add_driver_entry(ErlDrvEntry *de, DE_Handle *handle,
     }
 
     if (!err) {
-        err = init_driver(dp, de, handle);
+        err = init_driver(dp, de, handle, 0);
 
         if (taint) {
             erts_add_taint(dp->name_atom);
