diff --git a/erts/emulator/Makefile.in b/erts/emulator/Makefile.in
index a3c737bd9c9863566362cd70dd167d45a42806ae..e91c4c644b8a4a9c87f3e545054ed9cd6bf9d287 100644
--- a/erts/emulator/Makefile.in
+++ b/erts/emulator/Makefile.in
@@ -1162,7 +1162,6 @@ NIF_OBJS = \
 ifeq ($(TARGET),win32)
 DRV_OBJS = \
 	$(OBJDIR)/registry_drv.o \
-	$(OBJDIR)/inet_drv.o \
 	$(OBJDIR)/ram_file_drv.o
 OS_OBJS = \
 	$(OBJDIR)/dll_sys.o \
@@ -1189,7 +1188,6 @@ OS_OBJS = \
 	$(OBJDIR)/unix_socket_syncio.o
 
 DRV_OBJS = \
-	$(OBJDIR)/inet_drv.o \
 	$(OBJDIR)/ram_file_drv.o
 endif
 
diff --git a/erts/preloaded/src/prim_inet.erl b/erts/preloaded/src/prim_inet.erl
index 78c415e7c5240c8582aec89046d254e135c8eab8..fb15d8f5cc6caad25c636b742afa0da0f7815ebc 100644
--- a/erts/preloaded/src/prim_inet.erl
+++ b/erts/preloaded/src/prim_inet.erl
@@ -94,44 +94,8 @@ fdopen(Protocol, Family, Type, Fd, Bound)
     open(Protocol, Family, Type, [], ?INET_REQ_FDOPEN,
          [?int32(Fd), enc_value_2(bool, Bound)]).
 
-open(Protocol, Family, Type, Opts, Req, Data) ->
-    Drv = protocol2drv(Protocol),
-    AF = enc_family(Family),
-    T = enc_type(Type),
-    try erlang:open_port({spawn_driver,Drv}, [binary]) of
-	S ->
-	    case setopts(S, Opts) of
-		ok ->
-		    case ctl_cmd(S, Req, [AF,T,Data]) of
-			{ok,_} -> {ok,S};
-			{error,_}=E1 ->
-			    close(S),
-			    E1
-		    end;
-		{error,_}=E2 ->
-		    close(S),
-		    E2
-	    end
-    catch
-	%% The only (?) way to get here is to try to open
-	%% the sctp driver when it does not exist (badarg)
-	error:badarg       -> {error, eprotonosupport};
-	%% system_limit if out of port slots
-	error:system_limit -> {error, system_limit}
-    end.
-
-enc_family(inet)  -> ?INET_AF_INET;
-enc_family(inet6) -> ?INET_AF_INET6;
-enc_family(local) -> ?INET_AF_LOCAL.
-
-enc_type(stream) -> ?INET_TYPE_STREAM;
-enc_type(dgram) -> ?INET_TYPE_DGRAM;
-enc_type(seqpacket) -> ?INET_TYPE_SEQPACKET.
-
-protocol2drv(tcp)   -> "tcp_inet";
-protocol2drv(udp)   -> "udp_inet";
-protocol2drv(sctp)  -> "sctp_inet";
-protocol2drv(mptcp) -> "tcp_inet".
+open(_Protocol, _Family, _Type, _Opts, _Req, _Data) ->
+    {error, eprotonosupport}.
 
 drv2protocol("tcp_inet")  -> tcp;
 drv2protocol("udp_inet")  -> udp;
