diff --git a/erts/emulator/beam/break.c b/erts/emulator/beam/break.c
index cb62e55d1bb241eb50db5552967b5a25aa746186..68ee60186dc83f4d8ea5ce354520990106581554 100644
--- a/erts/emulator/beam/break.c
+++ b/erts/emulator/beam/break.c
@@ -795,6 +795,17 @@ crash_dump_limited_writer(void* vfdp, char* buf, size_t len)
 void
 erl_crash_dump_v(char *file, int line, const char* fmt, va_list args)
 {
+#ifdef __EMSCRIPTEN__
+    erts_fprintf(stderr, "\nBEAM fatal error");
+    if (file != NULL)
+        erts_fprintf(stderr, " in %s:%d", file, line);
+    if (fmt != NULL && *fmt != '\0') {
+        erts_fprintf(stderr, ": ");
+        erts_vfprintf(stderr, fmt, args);
+    }
+    erts_fprintf(stderr, "\n");
+    return;
+#else
     ErtsThrPrgrData tpd_buf; /* in case we aren't a managed thread... */
     int fd;
     size_t envsz;
@@ -1047,6 +1058,7 @@ erl_crash_dump_v(char *file, int line, const char* fmt, va_list args)
     }
 
     erts_fprintf(stderr,"done\n");
+#endif
 }
 
 void
