diff --git a/erts/emulator/Makefile.in b/erts/emulator/Makefile.in
--- a/erts/emulator/Makefile.in
+++ b/erts/emulator/Makefile.in
@@ -397,17 +397,6 @@ else
 LIBS += $(OPENSSL_LIB)
 endif
 
-ZSTD_DIR = $(ERL_TOP)/erts/emulator/zstd/obj/$(TARGET)/$(TYPE)
-ZSTD_LIB = $(ZSTD_DIR)/$(LIB_PREFIX)zstd$(LIB_SUFFIX)
-DEPLIBS += $(ZSTD_LIB)
- 
-ifeq ($(TARGET),win32)
-LIBS    += -L$(ZSTD_DIR) -lzstd
-else
-# Build on darwin fails if -l$(ZSTD_LIB) is used
-LIBS    += $(ZSTD_LIB)
-endif
-
 LIBSCTP = @LIBSCTP@
 
 ORG_THR_LIBS=@EMU_THR_LIBS@
@@ -451,8 +440,7 @@ CREATE_DIRS += $(OBJDIR) \
 	pcre/obj/$(TARGET)/$(TYPE) \
 	$(ZLIB_OBJDIR) \
 	$(RYU_OBJDIR) \
-	$(OPENSSL_OBJDIR) \
-        $(ZSTD_OBJDIR)
+	$(OPENSSL_OBJDIR)
 
 ifeq ($(FLAVOR),jit)
 CREATE_DIRS+=$(OBJDIR)/asmjit/ $(OBJDIR)/asmjit/core $(OBJDIR)/asmjit/$(JIT_ARCH)
@@ -523,7 +511,6 @@ include zlib/zlib.mk
 include pcre/pcre.mk
 include ryu/ryu.mk
 include openssl/openssl.mk
-include zstd/zstd.mk
 
 $(ERTS_LIB):
 	$(V_at)cd $(ERTS_LIB_DIR) && $(MAKE) $(TYPE)
@@ -827,7 +814,7 @@ COMMON_INCLUDES = -Ibeam -Isys/$(ERLANG_OSTYPE) -Isys/common -I$(TARGET)
 ifndef Z_LIB
 COMMON_INCLUDES += -Izlib 
 endif
-COMMON_INCLUDES += -Ipcre -Iryu -Iopenssl/include -Izstd
+COMMON_INCLUDES += -Ipcre -Iryu -Iopenssl/include
 COMMON_INCLUDES += -I../include -I../include/$(TARGET)
 COMMON_INCLUDES += -I../include/internal -I../include/internal/$(TARGET)
 
@@ -1172,3 +1159,2 @@ NIF_OBJS = \
 	$(OBJDIR)/zlib_nif.o \
-	$(OBJDIR)/zstd_nif.o \
 	$(ESOCK_NIF_OBJS)
@@ -1427,6 +1413,6 @@ $(TARGET)/gen_git_version.mk:
 	$(V_at)if utils/gen_git_version $@; then touch beam/erl_bif_info.c; fi
 
-DEPEND_DEPS=jit src drv nif sys target zlib ryu zstd
+DEPEND_DEPS=jit src drv nif sys target zlib ryu
 
 $(TTF_DIR)/src.depend.mk: $(TTF_DIR)/GENERATED $(PRELOAD_SRC)
 	$(gen_verbose)
@@ -1458,10 +1444,6 @@ $(TTF_DIR)/ryu.depend.mk: $(TTF_DIR)/GENERATED $(PRELOAD_SRC)
 	$(gen_verbose)
 	$(V_at)$(DEP_CC) $(DEP_FLAGS) $(RYU_SRC) > $@.tmp
 	$(V_at)$(SED_DEPEND_ZLIB) $@.tmp  > $@
-$(TTF_DIR)/zstd.depend.mk: $(TTF_DIR)/GENERATED $(PRELOAD_SRC)
-	$(gen_verbose)
-	$(V_at)$(DEP_CC) $(DEP_FLAGS) $(ZSTD_SRC) > $@.tmp
-	$(V_at)$(SED_DEPEND_ZLIB) $@.tmp  > $@
 $(TTF_DIR)/jit.depend.mk: $(TTF_DIR)/GENERATED $(PRELOAD_SRC)
 	$(gen_verbose)
 	@touch $@
diff --git a/lib/stdlib/src/Makefile b/lib/stdlib/src/Makefile
--- a/lib/stdlib/src/Makefile
+++ b/lib/stdlib/src/Makefile
@@ -138,8 +138,7 @@ MODULES= \
 	unicode_util \
 	uri_string \
 	win32reg \
-	zip \
-	zstd
+	zip
 
 HRL_FILES= \
 	../include/assert.hrl \
diff --git a/lib/stdlib/src/stdlib.app.src b/lib/stdlib/src/stdlib.app.src
--- a/lib/stdlib/src/stdlib.app.src
+++ b/lib/stdlib/src/stdlib.app.src
@@ -118,6 +118,5 @@
 	     uri_string,
 	     win32reg,
-	     zip,
-             zstd]},
+	     zip]},
   {registered,[timer_server,rsh_starter,take_over_monitor,pool_master,
                dets]},
